### PR TITLE
add `id` and `env` options to `process`

### PR DESCRIPTION
**Pull request by dstockhammer**
# One-line summary

Allow filtering the `process` operation by id and env

## Types of Changes

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: clin/run.py [hunk 1]
```diff
@@ -99,24 +139,42 @@ def apply(
...
+    multiple=True,
+    help="Select one or multiple steps to process by their id",
+)
+@click.option(
+    "-e",
+    "--env",
+    required=False,
+    type=str,
+    multiple=True,
+    help="Select one or multiple steps to process by mathing the target environment",
```
Comment (Dmitry-Erokhin) [inline clin/run.py:156]:
```suggestion
    help="Select one or multiple steps to process by matching the target environment",
```

File: clin/run.py [hunk 2]
```diff
@@ -99,24 +139,42 @@ def apply(
...
 @click.argument("file", type=click.Path(exists=True, dir_okay=False, readable=True))
 def process(
-    token: Optional[str],
-    execute: bool,
+    token: str,
     verbose: bool,
+    execute: bool,
     show_diff: bool,
     show_payload: bool,
+    id: Tuple[str],
```
Comment (Dmitry-Erokhin) [inline clin/run.py:165]:
```suggestion
    ids: Tuple[str],
```

File: clin/run.py [hunk 3]
```diff
@@ -99,24 +139,42 @@ def apply(
...
+    env: Tuple[str],
```
Comment (Dmitry-Erokhin) [inline clin/run.py:166]:
```suggestion
    envs: Tuple[str],
```

File: clin/run.py (hunk 2 above)
Comment (dstockhammer) [reply clin/run.py:165]: Not sure about this one. The command syntax is like this:

```
clin apply --id xxx event.yaml
clin apply --id xxx --id yyy --id zzz event.yaml
```

I'm not sure if there is a commonly accepted pattern for this, but to me if it was `--ids` I would probably expect to be able to pass a csv or something like that. Maybe like `--ids xxx,yyy,zzz`. For what it's worth, the [click docs](https://click.palletsprojects.com/en/7.x/options/#multiple-options) also uses singular.

File: clin/run.py (hunk 3 above)
Comment (dstockhammer) [reply clin/run.py:166]: see above

File: clin/run.py (hunk 2 above)
Comment (Dmitry-Erokhin) [reply clin/run.py:158]: I'm just saying, that variable that refers to the collection should be in plural form.
It is not about cli API, it is about function params naming only.

File: clin/run.py (hunk 3 above)
Comment (Dmitry-Erokhin) [reply clin/run.py:159]: the same

File: clin/run.py (hunk 2 above)
Comment (dstockhammer) [reply clin/run.py:158]: Oh ok, I see, that would make sense. Unfortunately that not possible though; the arg names must match the click API

Comment (Dmitry-Erokhin) [reply clin/run.py:158]: Ah! Indeed. Sorry for the confusion.

**Comment by Dmitry-Erokhin** [comment]
👍

**Pull request merged**